### ISSUE TITLE
Train and assess performance of YOLOv8 object detection model

### ISSUE THREAD

**Issue by OwenLeSann** [mcgill-robotics/AUV-2026#10]
## 🎯 Objective
The objective of this task is to first ensure that the AUV's down and front camera YOLO object detection models can be trained on the Jetson Orin Nano in the ROS2 workspace, and secondly, to test the model's current validity and highlight what changes must be made to improve the performance of the model.

## 📖 Description
The current YOLO model performs well in the sim but poorly in real conditions. Before exploring alternative models or image enhancements, we must determine what is causing the lacklustre performance in our current model. The goal of this issue is to:

- [ ] Ensure that the training.py and YOLO_AUV_Training_Notebook.ipynb training programs can be executed with CUDA acceleration on the Jetson Orin Nano in the ROS2 workspace.

- [ ] Model training and validation losses over time to test if our model, with its current optimizer and learning rate schedule, converges to the global optimum within 200 epochs (our current training time)

- [ ] Model training and validation losses over time to test our model for overfitting or underfitting

- [ ] Inspect bounding box validation losses to assess the model's ability to place and size bounding boxes accurately around detected objects

- [ ] Inspect classification validation losses to assess the model's ability to correctly classify objects within the predicted bounding boxes

- [ ] Inspect distribution focal validation losses to assess models' robustness against class imbalances

- [ ] Inspect other YOLO performance metrics collected by Ultralytics, including, importantly:
  - [ ] F1 score curve to assess the model's balance between false positives and false negatives
  - [ ] Precision-recall curve to assess trade-offs between precision (the accuracy of the detected objects, indicating how many detections were correct) and recall (the ability of the model to identify all instances of objects in the images)
  - [ ] mAP50-95 to assess, comprehensively, the model's performance across different levels of detection difficulty

- [ ] Finally, test and report the model's accuracy against the unseen test set (after all of the above validations are complete)

- [ ] After completing the above on the RoboFlow 2024 image database and object classes, repeat with the RoboFlow 2025 image dataset

## ✅ Acceptance Criteria
Step by step above

## 🛠️ Implementation
**Approach:** 
**Dependencies:** 

## 🔗 Links
**Related:** None
**Blocks:** None
**Blocked by:** pip is currently malfunctioning in the ros2_ws Docker container, which prevents the training script from downloading dependencies; No issue posted
**Future work:** Tune YOLO models and possibly enhance training images
